pkgs/development/python-modules/expecttest/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+# Native build inputs
+, poetry-core
+
+# Check inputs
+, python
+, hypothesis
+}:
+
+buildPythonPackage rec {
+  pname = "expecttest";
+  version = "0.1.3";
+  format = "pyproject";
+
+  # Pypi doesn't contain the test, so fetch from GitHub
+  src = fetchFromGitHub {
+    owner = "ezyang";
+    repo = "expecttest";
+    rev = "4be7413c98b4c6bdb4b26bfc8358c0a909c8d675";
+    sha256 = "sha256:051bffhfii5vqhfpgvcl5hkbj7c24p4j93hggmhf5k92rfdcd16i";
+  };
+
+  nativeBuildInputs = [ poetry-core ];
+
+  checkInputs = [ hypothesis ];
+
+  checkPhase = "${python.interpreter} test_expecttest.py";
+
+  pythonImportsCheck = [ "expecttest" ];
+
+  meta = with lib; {
+    description = ''This library implements expect tests (also known as "golden" tests).'';
+    homepage = "github.com/ezyang/expecttest";
+    license = licenses.mit;
+    maintainers = with maintainers; [  ];
```

Comment:
New packages needs a maintainer.

Reply:
I'm unlikely to be a responsive maintainer (sorry about that!). 

I've left this as a draft PR for now in case anyone else wants this packaged merged though.